### PR TITLE
[api] refactor web3 test

### PR DESCRIPTION
**Pull request by Liuhaai**

### CODE REVIEW

**Review by CoderZhi — APPROVED** [review]

File: api/web3server_test.go [hunk 1]
```diff
@@ -313,35 +306,30 @@ func TestEstimateGas(t *testing.T) {
 		},
 	}
 
-	for _, v := range testData {
-		ret, err := svr.web3Server.estimateGas(v.input)
-		require.NoError(err)
-		require.Equal(ret, uint64ToHex(v.result))
+	for i, v := range testData {
+		t.Run(fmt.Sprintf("%d-%d", i, len(testData)-1), func(t *testing.T) {
```
Comment (dustinxie) [inline api/web3server_test.go:310]: IMO we should not use `t.Run` here inside for loop? only 3 lines of code, not worth it

`t.Run` is meant to run the entire test with a different config, like
```
testIndexer := func(kvStore db.KVStore, t *testing.T) {
   // 100 lines of test code
}
t.Run("in memory", func(t *testing.T) {
	testIndexer(db.NewMemKVStore(), t)
})
t.Run("db files", func(t *testing.T) {
	testIndexer(db.NewBoltDB(), t)
})
```

File: api/web3server_test.go [hunk 2]
```diff
@@ -21,50 +21,56 @@ import (
...
-		}
-		_, err := parseWeb3Reqs(test.req)
-
-		if test.hasError {
-			require.Error(err)
-		} else {
-			require.NoError(err)
-		}
+	for _, test := range testData {
+		t.Run(test.testName, func(t *testing.T) {
```
Comment (dustinxie) [inline api/web3server_test.go:63]: IMO we should not use `t.Run` here inside for loop? only couple of lines of code, not worth it

`t.Run` is meant to run the entire test with a different config, like
```
testIndexer := func(kvStore db.KVStore, t *testing.T) {
   // 100 lines of test code
}
t.Run("in memory", func(t *testing.T) {
	testIndexer(db.NewMemKVStore(), t)
})
t.Run("db files", func(t *testing.T) {
	testIndexer(db.NewBoltDB(), t)
})
```

File: api/web3server_test.go [hunk 3]
```diff
@@ -175,94 +164,98 @@ func TestGetBlockByNumber(t *testing.T) {
...
-		require.NoError(err)
-		if v.expected == 0 {
-			require.Nil(ret)
-			continue
-		}
-		blk, ok := ret.(blockObject)
-		require.True(ok)
-		require.Equal(len(blk.Transactions), v.expected)
+	for i, v := range testData {
+		t.Run(fmt.Sprintf("%d-%d", i, len(testData)-1), func(t *testing.T) {
```
Comment (dustinxie) [inline api/web3server_test.go:168]: same here

File: api/web3server_test.go [hunk 4]
```diff
@@ -175,94 +164,98 @@ func TestGetBlockByNumber(t *testing.T) {
...
+	}{
+		{[]interface{}{"0xDa7e12Ef57c236a06117c5e0d04a228e7181CF36", "0x1"}, 2},
+		{[]interface{}{"0xDa7e12Ef57c236a06117c5e0d04a228e7181CF36", "pending"}, 2},
+	}
 
-	testData2 := []interface{}{"0xDa7e12Ef57c236a06117c5e0d04a228e7181CF36", "pending"}
-	ret, _ = svr.web3Server.getTransactionCount(testData2)
-	require.Equal(ret, uint64ToHex(2))
+	for i, v := range testData {
+		t.Run(fmt.Sprintf("%d-%d", i, len(testData)-1), func(t *testing.T) {
```
Comment (dustinxie) [inline api/web3server_test.go:207]: same here

File: api/web3server_test.go (hunk 2 above)
Comment (Liuhaai) [reply api/web3server_test.go:63]: Different data is tested in the loop. With `t.run`, it's easier to trace where the err happens

Comment (dustinxie) [reply api/web3server_test.go:63]: better keep the coding style consistency in test code

**Review by dustinxie — APPROVED** [review]

**Pull request merged**